README.md

```diff
@@ -29,30 +29,116 @@ These public reports help protect the open source community, and provide a data
 - vulnerability reports
 - compromised infrastructure
-- non-malicious packages
+- offensive security tools, unless they execute malicious payloads on install
 
-### Prior Work
+## Definition of a Malicious Package
 
-- GitHub's [Advisory Database (filtered by malware)](https://github.com/advisories?query=type%3Amalware), for the NPM ecosystem.
-- https://github.com/lxyeternal/pypi_malregistry (PyPI)
-- https://dasfreak.github.io/Backstabbers-Knife-Collection/ (PyPI and npm), by Marc Ohm et al.
-- https://github.com/datadog/malicious-software-packages-dataset (PyPI), by Datadog
+Below is the definition of what this repository considers a malicious package.
+
+- a package publicly available in a package registry
+- and either:
+  - when installed or used, would require some sort of incident response due to
+    the loss of confidentiality, availability and/or integrity; or
+  - exfiltrates an identifier that can be directly used to launch a subsequent 
+    attack against the victim (e.g. username for phishing or password
+    bruteforcing)
+- and also either:
+  - violates the terms of the package registry; or
+  - would be reasonably considered to require removal from the package registry
+
+### Dependency and manifest confusion
+
+[Dependency confusion](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610)
+and [manifest confusion](https://blog.vlt.sh/blog/the-massive-hole-in-the-npm-ecosystem)
+are techniques that exploit quirks in the behavior of package systems and how
+they are used within organizations. Packages using these attacks are malicious.
+
+Very occasionally someone may unintentionally encounter these quirks, but
+this is considered infrequent.
+
+Manifest confusion requires someone to bypass the NPM command line tool and
+deliberately provide an altered manifest.
+
+Dependency confusion are effectively the same as an account takeover where an
+attacker replaces a package's code with their own. This means even trivial or
+empty dependency confusion packages would require incident response.
+
+### Spam and typosquating
+
+Spam, typosquatting are not malicious, unless the package itself exhibits
+malicious behavior as-per the definition above.
+
+These types of packages are often empty (i.e. no functional code), or consist of
+only useless trivial functionality (e.g. printing a message). While these
+packages are not malicious, they are a nuisance and generally unwanted.
+
+Typosquatting packages may be hard to distinguish from dependency confusion. As
+a result, these reports are allowed to be present in the malicious packages
+repository.
+
+### Reverse engineering protection (e.g. obfuscation)
+
+Reverse engineering protections are not malicious, unless it exhibits malicious
+behavior as-per the definition above.
+
+Obfuscation, debugger evasion, and other reverse engineering protection
+techniques, are used by both developers seeking to protect their source code
+and attackers seeking to evade detection.
```

Comment:
The crates.io usage policy forbids any content that

> uses obfuscation to hide or mask functionality

which seems to suggest to me that even using obfuscation for protecting source code is considered unacceptable by crates.io, not just if the obfuscation is done to hide malicious behavior.